cmake/Dependencies.cmake

```diff
@@ -33,6 +33,50 @@ macro(enable_ubsan)
   endif()
 endmacro()
 
+# ---[ CUDA
+if(USE_CUDA)
```

Comment:
This code is the same, but moved to the top because `thread.cmake` uses `USE_CUDA` now.